evalml/data_checks/datetime_format_data_check.py

```diff
@@ -118,21 +151,52 @@ def validate(self, X, y):
             )
             return messages
 
+        is_increasing = pd.DatetimeIndex(datetime_values).is_monotonic_increasing
         if not inferred_freq:
             col_name = (
                 self.datetime_column
                 if self.datetime_column != "index"
                 else "either index"
             )
-            messages.append(
-                DataCheckError(
-                    message=f"No frequency could be detected in {col_name}, possibly due to uneven intervals.",
-                    data_check_name=self.name,
-                    message_code=DataCheckMessageCode.DATETIME_HAS_UNEVEN_INTERVALS,
-                ).to_dict()
-            )
-
-        if not (pd.DatetimeIndex(datetime_values).is_monotonic_increasing):
+            frequencies = []
+            missing = []
+            for i in range(len(datetime_values) - 2):
+                freq = pd.infer_freq(datetime_values[i : i + 3])
+                if freq is None and i not in missing:
+                    missing.append(i + 1)
```

Comment:
@fjlanasa Do we need to return the missing indices? It could make the response very big since there could potentially be many missing dates.

Reply:
I think we can do without